datafusion/datasource/src/file_scan_config.rs

```diff
@@ -174,6 +175,219 @@ pub struct FileScanConfig {
     pub batch_size: Option<usize>,
 }
 
+#[derive(Clone)]
+pub struct FileScanConfigBuilder {
```

Comment:
Can we please add comments and examples here? Perhaps just pointing back to `FileScanConfig`?